packages/types-core/src/project/versioned/v1_0_0/types.ts

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020-2023 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {BaseDataSource, BaseTemplateDataSource} from '../base';
+
+/**
+ * Represents specifications for an indexer runner.
+ * @interface
+ */
+export interface RunnerSpecs {
+  /**
+   * The specifications for a node.
+   * @type {NodeSpec}
+   */
+  node: NodeSpec;
+
+  /**
+   * The specifications for a query.
+   * @type {QuerySpec}
+   */
+  query: QuerySpec;
+}
+
+/**
+ * Represents specifications for a node.
+ * @interface
+ * @example
+ * // Example of a substrate based NodeSpec
+ * const nodeSpec = {
+ *   name: '@subql/node'
+ *   version: '^2.15.0'
+ *   options: {
+ *     historical: true
+ *   }
+ * }
+ */
+export interface NodeSpec {
+  /**
+   * The name of the indexer node.
+   * @type {string}
+   */
+  name: string;
+
+  /**
+   * The version number of the indexer node.
+   * @type {string}
+   * @default "*"
+   */
+  version: string;
+
+  /**
+   * Additional options for the node (optional).
+   * @type {NodeOptions}
+   */
+  options?: NodeOptions;
+}
+
+/**
+ * Represents specifications for a query.
+ * @interface
+ */
+export interface QuerySpec {
+  /**
+   * The name of the query service.
+   * @type {string}
+   * @default "@subql/query"
+   */
+  name: string;
+  /**
+   * The version number of the query service.
+   * @type {string}
+   * @default "*"
+   */
+  version: string;
+}
+
+/**
+ * Represents optional configuration options for a node.
+ * @interface
+ */
+export interface NodeOptions {
+  /**
+   * Indicates whether historical state tracking is enabled for the indexer (optional).
+   * @type {boolean}
+   * @default "true"
```

Comment:
Is this default correct? I thought we defaulted to enabling historical?

Reply:
Yes, historical is enabled by default